finesse/hardware/temperature/dp9800.py

```diff
@@ -0,0 +1,238 @@
+        sysflag: string representation of the system flag bitmask
+    """
+    check_data(data)
+    bcc = calculate_bcc(data)
+    if bcc != data[-2]:
+        raise DP9800Error("BCC check failed")
+
+    try:
+        data_ascii = data.decode("ascii")
+    except UnicodeDecodeError as e:
+        raise DP9800Error(e)
+
+    vals_begin = 3
+    vals_end = 74  # assuming 9 temperatures
+    etx_index = data.find(b"\x03")
+
+    vals = [Decimal(val) for val in data_ascii[vals_begin:vals_end].split()]
+
+    sysflag = bin(int(data_ascii[vals_end:etx_index], 16))
+
+    # Omit mysterious first temperature, and omit binary identifier characters
+    return vals[1:], sysflag[2:]
+
+
+class DP9800Error(Exception):
+    """Indicates that an error occurred while communicating with the device."""
+
+
+class DP9800(TemperatureMonitorBase):
+    """An interface for DP9800 temperature readers.
+
+    The manual for this device is available at:
+    https://assets.omega.com/manuals/M5210.pdf
+    """
+
+    def __init__(self, serial: Serial) -> None:
+        """Create a new DP9800 from an existing serial device.
+
+        Args:
+            serial: Serial device
+        """
+        super().__init__("DP9800")
+        self.serial = serial
+
+    def close(self) -> None:
+        """Close the connection to the device.
+
+        Raises:
+            SerialException: Error communicating with device
+        """
+        try:
+            self.serial.close()
+        except SerialException as e:
+            raise DP9800Error(e)
+        else:
+            pub.sendMessage("temperature_monitor.close")
+            logging.info("Closed connection to DP9800")
+
+    def get_device_settings(self, sysflag: str) -> dict[str, str]:
+        """Provide the settings of the device as stored in the system flag.
+
+        The system flag is stored as a bit mask with the format TxxLxSAF,
+        where:
+            F - bit 0: temperature unit: 0 = deg C, 1 = deg F
+            A - bit 1: audible button:   0 = silence, 1 = audible
+            S - bit 2: autoscan:         0 = no scan, 1 = autoscan active
+            x - bit 3: must be 0
+            L - bit 4: logging:          0 = no logging, 1 = logging active
+            x - bit 5: must be 0
+            x - bit 6: must be 0
+            T - bit 7: instrument type:  0 = TC, 1 = PT
+
+        Args:
+            sysflag: string representation of the system flag bitmask
+
+        Returns:
+            dev_settings: dictionary containing the current device settings
+        """
+        dev_settings = {
+            "instrument_type": ["TC", "PT"][int(sysflag[0])],
+            "logging_state": ["no logging", "logging active"][int(sysflag[3])],
+            "scanning_state": ["no scan", "autoscan active"][int(sysflag[5])],
+            "audible_state": ["silence", "audible"][int(sysflag[6])],
+            "temperature_unit": ["deg C", "deg F"][int(sysflag[7])],
+        }
+        return dev_settings
```

Comment:
Much better! Small suggestion:
```suggestion
        Returns:
            dictionary containing the current device settings
        """
        return {
            "instrument_type": ["TC", "PT"][int(sysflag[0])],
            "logging_state": ["no logging", "logging active"][int(sysflag[3])],
            "scanning_state": ["no scan", "autoscan active"][int(sysflag[5])],
            "audible_state": ["silence", "audible"][int(sysflag[6])],
            "temperature_unit": ["deg C", "deg F"][int(sysflag[7])],
        }
```